// @socialgouv/react-departements/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactSvgmt = require("react-svgmt");

var _Carte = _interopRequireDefault(require("./Carte"));

var _departements = require("./departements");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance"); }

function _iterableToArray(iter) { if (Symbol.iterator in Object(iter) || Object.prototype.toString.call(iter) === "[object Arguments]") return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = new Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } }

var France = function France(_ref) {
  var color = _ref.color,
      highlightColor = _ref.highlightColor,
      departements = _ref.departements;
  var dpts = [].concat(_toConsumableArray(departements), _toConsumableArray((0, _departements.hasPetiteCouronne)(departements) ? ["75-92-93-94"] : []));
  return _react["default"].createElement(_reactSvgmt.SvgLoader, {
    svgXML: _Carte["default"]
  }, _react["default"].createElement(_reactSvgmt.SvgProxy, {
    selector: "#carte",
    fill: color
  }), dpts.map(function (dpt) {
    return _react["default"].createElement(_reactSvgmt.SvgProxy, {
      key: dpt,
      selector: "#dpt-".concat((0, _departements.normalizeDpt)(dpt)),
      fill: highlightColor
    });
  }));
};

France.propTypes = {
  color: _propTypes["default"].string,
  departements: _propTypes["default"].array,
  highlightColor: _propTypes["default"].string
};
France.defaultProps = {
  color: "#74B4FF",
  highlightColor: "#b3ff75",
  departements: []
};
var _default = France;
exports["default"] = _default;
